discovery/kubernetes/kubernetes.go

```diff
@@ -256,7 +258,13 @@ func New(l log.Logger, conf *SDConfig) (*Discovery, error) {
 		kcfg *rest.Config
 		err  error
 	)
-	if conf.APIServer.URL == nil {
+	if conf.KubeConfig != "" {
+		kcfg, err = clientcmd.BuildConfigFromFlags("", conf.KubeConfig)
```

Comment:
Is there a way to have metrics for that client? With go-conntrack?

Reply:
no ,this step is load Kube config file and create an rest.Config, it won't create any communicate in the buildConfig step

Reply:
That is fine, we just have to ensure this is an exception in unmarshal. There is already such an exception (see previous comment)